selectors/selectors.md

```diff
@@ -113,19 +113,23 @@ These components imply or expand into the following things:
 Selector types
 --------------
 
-### CID Selector
+### Path Selector
 
-Selects a single CID.
+Selects a single path segment.
 
 
-### Path Selector
+### Array Selectors
 
-Selects a single path segment.
+There's three selectors that operate on arrays:
+
+ - Array All Selector: selects all items of an array
+ - Array Slice Selector: selects a given range of an array
+ - Array Position Selector: selects a single element from an array
```

Comment:
> If we're trying to do type checking _and_ path selection, I think this is the wrong place to do so.

Yes, this was the intention. I would expect that the one writing the query knows whether it's a map or an array and would rather prefer getting an error message than weird results.

Reply:
I don't really see how the results will be weird. We're basically just selecting different paths through the graph. If we select a subset of an array, we get a subset of the array.

Reply:
And if it isn't an array, you would get nothing back instead of an error.

Well, perhaps the implementations could be smart enough that if you want to do a subset and it's not an array it will give you an error. But then again, do you always want an error in that case or just skip it?